merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/GetSimulationTemplateAction.java

```
@@ -28,13 +27,11 @@ public GetSimulationTemplateAction(final Connection connection) throws SQLExcept
     this.statement = connection.prepareStatement(sql);
   }
 
-  public Optional<SimulationTemplateRecord> get(final long simulationTemplateId)
+  public SimulationTemplateRecord get(final long simulationTemplateId)
```

If it's no longer safe to call this action on a non-existent simulation template id - can we add a javadoc comment documenting what it will do if it is provided a non-existent id? It is true that it is currently called in a safe fashion, but by changing the behavior from "returns Optional.empty()" to "throws an exception, probably?", we've made it more difficult for a future programmer to use this action safely.

---

Done 👍. Also added a similar documentation for GetSimulationAction, where the same change was made in a prior PR.

Adjacently, I'm not sure if this method is now more "unsafe". Obviously it throws an exception rather than returning an empty Optional if there are no results, but blindly using that Optional without checking if the value is present would already lead to an exception (with a slightly obfuscated root cause, depending on how far from the `get` the value is blindly used), and I'm not certain the safety differences between checking the Optional as soon as it's returned and throwing an exception if it's empty and throwing an exception in the method itself. Could you please clarify that part?

---

Resolving as I've dropped the commit this conversation is about.